extensions/amp-story/1.0/amp-story.js

```diff
@@ -457,12 +458,14 @@ export class AmpStory extends AMP.BaseElement {
         return;
       }
 
-      this.switchTo_(e.detail.targetPageId);
+      this.switchTo_(getDetail(e)['targetPageId']);
       this.ampStoryHint_.hideAllNavigationHint();
     });
 
     this.element.addEventListener(EventType.PAGE_PROGRESS, e => {
-      const {pageId, progress} = e.detail;
+      const detail = getDetail(e);
```

Comment:
Since this is such common syntax, we should create a lint rule to ensure new contributors understand why this pattern exists.